developer-docs-site/static/messageboard_demo/client/README.md

```diff
@@ -0,0 +1,187 @@
+```
+
+**This demo shows**
+
+1. write a basic smart contract with move [[source](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/messageboard)]
+2. publish, create and interact with messageboard [[source](https://github.com/aptos-labs/aptos-core/tree/main/developer-docs-site/static/messageboard_demo/client)]
+
+A messageboard has an admin who is the creator of the messageboard.
+
+The messageboard has a pinned_post resource. Only participants who have the authority can modify this pinned message resource. Admin can provide this authority to the participants.
+
+Besides the pinned_post resource, anyone can send a message to the messageboard. this will trigger a MessageChangeEvent. Anyone can subscribe to the event stream to read all the messages sent.
+
+## Messageboard Smart Contract
+
+There are two types of messageboards: capability-based messageboard and ACL-based messageboard. The capability-based messageboard is generally more secure [eg: [confused deputy problem](https://en.wikipedia.org/wiki/Confused_deputy_problem)] and also prevents the problem of doing ACL validation against a long list.
+
+The source code can be found here [[source](https://github.com/aptos-labs/aptos-core/tree/main/aptos-move/move-examples/messageboard)]
+
+### Capability-Based Messageboard
+
+This [messageboard](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/move-examples/messageboard/sources/CAPMessageBoard.move) uses MessageChangeCapability to control who can change the pinned_post.
+
+```tsx
+struct MessageChangeCapability has key, store {
+    board: address
+}
+```
+
+When a participant wants to modify the pinned post. The participant first acquire his capability through borrowing his message change capability resource. We then examine if the cap’s board is the same board address that the participant is trying to change the pinned_post.
+
+```tsx
+public(script) fun send_pinned_message(
+        account: &signer, board_addr: address, message: vector<u8>
+    ) acquires MessageChangeCapability, MessageChangeEventHandle, CapBasedMB {
+        let cap = borrow_global<MessageChangeCapability>(Signer::address_of(account));
+        assert!(cap.board == board_addr, EACCOUNT_NO_NOTICE_CAP);
+        let board = borrow_global_mut<CapBasedMB>(board_addr);
+        board.pinned_post = message;
+       ...
+    }
+```
+
+To provide participant the capability resource, the board creator creates the MessageChangeCapability and offer the capability to a participant’s address. 
+
+```tsx
+public(script) fun add_participant(account: &signer, participant: address) acquires MessageCapEventHandle {
+    let board_addr = Signer::address_of(account);
+    Offer::create(account, MessageChangeCapability{ board: board_addr }, participant);
+...
+}
+```
+
+The participant can then claim this capability and then move it to their own account. Later on, the participant can use this resource stored in their account to modify the pinned_post
```

Comment:
The participant can then claim this capability and move it to their own account. Later on, the participant can use this resource stored in their account to modify the pinned_post.

Reply:
updated